File: comparison/results/haiku/code-simplifier/humaneval/HumanEval_1/python/solution.py
from typing import List


def separate_paren_groups(paren_string: str) -> List[str]:
    """Input to this function is a string containing multiple groups of nested parentheses. Your goal is to
    separate those groups into separate strings and return the list of those.
    Separate groups are balanced (each open brace is properly closed) and not nested within each other.
    Ignore any spaces in the input string.
    >>> separate_paren_groups('( ) (( )) (( )( ))')
    ['()', '(())', '(()())']
    """
    clean_string = paren_string.replace(' ', '')
    
    result = []
    current_group = []
    balance = 0
    
    for char in clean_string:
        current_group.append(char)
        if char == '(':
            balance += 1
        elif char == ')':
            balance -= 1
        
        if balance == 0 and current_group:
            result.append(''.join(current_group))
            current_group = []
    
    return result

File: comparison/results/haiku/code-simplifier/humaneval/HumanEval_1/javascript/solution.js
function separateParenGroups(parenString) {
  const cleanString = parenString.replace(/\s/g, '');
  const result = [];
  let currentGroup = '';
  let balance = 0;
  
  for (const char of cleanString) {
    currentGroup += char;
    
    if (char === '(') {
      balance++;
    } else if (char === ')') {
      balance--;
    }
    
    if (balance === 0) {
      result.push(currentGroup);
      currentGroup = '';
    }
  }
  
  return result;
}
